fractal_server/app/api/v1/workflow.py

```diff
@@ -227,18 +227,20 @@ async def patch_workflow(
     )
 
     for key, value in patch.dict(exclude_unset=True).items():
-        if key == "order_permutation":
-            if len(value) != len(workflow.task_list):
+        if key == "reordered_workflowtask_ids":
+            if len(value) != len(workflow.task_list) or set(value) != set(
```

Comment:
Maybe better to define the `current_worfklowtask_ids` variable, and use it twice (in the check and in the error message)

Reply:
Also: I think the second check is wrong (you are comparing with task_list itself, not with the IDs)